### ISSUE TITLE
go/parse: confusing error message caused by stray comma in struct field

### ISSUE THREAD

**Issue by josharian** [golang/go#10579]
Hit "Format" on http://play.golang.org/p/QX0D-qyFwJ

```go
package main

type T struct {
	a, int
}
```

Result: "prog.go:4:2: expected anonymous field"

**Comment by minux** [comment]
the error produced by gc is not good either.
main.go:4: syntax error: unexpected semicolon or newline

it took me some time to understand why it's complaining
about the newline (it thinks int as a field name, and expects
a type)

I think it's pretty hard to generate good diagnostics for this
case due to the ambiguity (it could be a missing field name
or a missing type name).

**Comment by mattn** [comment]
btw, `Run` says: `main.go:4: syntax error: unexpected semicolon or newline`

**Comment by josharian** [comment]
> I think it's pretty hard to generate good diagnostics for this case due to the ambiguity (it could be a missing field name or a missing type name).

True, although an error message saying "missing field name or type name" would be better than what we've got.

Not urgent, by any means.

**Comment by ianlancetaylor** [comment]
For the record, gccgo says

foo.go:4:11: error: expected type
